### PR TITLE
[ProductBundle] Fix default repository for sylius product association type

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Branch?         | 1.3
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | mentioned in #10350
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ProductBundle/test/src/Tests/SyliusProductBundleTest.php [hunk 1]
```diff
@@ -38,4 +43,26 @@ public function its_services_are_initializable(): void
...
+            'sylius.repository.product' => ProductRepository::class,
+            'sylius.repository.product_association_type' => ProductAssociationTypeRepository::class,
+            'sylius.repository.product_variant' => ProductVariantRepository::class,
+            'sylius.repository.product_attribute' => EntityRepository::class,
+            'sylius.repository.product_option' => ProductOptionRepository::class,
+        ];
+
+        foreach ($repositories as $id => $class) {
+            $service = $container->get($id, ContainerInterface::NULL_ON_INVALID_REFERENCE);
+            Assert::assertEquals(get_class($service), $repositories[$id]);
```
Comment (loic425) [inline src/Sylius/Bundle/ProductBundle/test/src/Tests/SyliusProductBundleTest.php]: My bad it should be $class

File: src/Sylius/Bundle/ProductBundle/test/src/Tests/SyliusProductBundleTest.php [hunk 2]
```diff
@@ -38,4 +43,26 @@ public function its_services_are_initializable(): void
             Assert::assertNotNull($container->get($id, ContainerInterface::NULL_ON_INVALID_REFERENCE));
         }
     }
+
+    /**
+     * @test
+     */
+    public function its_initializes_default_doctrine_repositories(): void
```
Comment (GSadee) [inline src/Sylius/Bundle/ProductBundle/test/src/Tests/SyliusProductBundleTest.php]: `it_initializes...`

**Comment by lchrusciel** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "fix/default-repositories"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

File: src/Sylius/Bundle/ProductBundle/test/src/Tests/SyliusProductBundleTest.php (hunk 2 above)
Comment (loic425) [reply src/Sylius/Bundle/ProductBundle/test/src/Tests/SyliusProductBundleTest.php]: it's now fixed.

**Comment by lchrusciel** [comment]
Can you rebase one more time to 1.3? #10386 has a fix for the build

**Comment by loic425** [comment]
> Can you rebase one more time to 1.3? #10386 has a fix for the build

Done

File: src/Sylius/Bundle/ProductBundle/Resources/config/services/integrations/doctrine/orm.xml [hunk 3]
```diff
@@ -14,6 +14,7 @@
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <parameters>
         <parameter key="sylius.repository.product.class">Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductRepository</parameter>
+        <parameter key="sylius.repository.product_association_type.class">Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductAssociationTypeRepository</parameter>
```
Comment (Zales0123) [inline src/Sylius/Bundle/ProductBundle/Resources/config/services/integrations/doctrine/orm.xml]: Shouldn't we then remove [this configuration](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Resources/config/app/sylius/sylius_product.yml#L23)? 🤔

Comment (loic425) [reply src/Sylius/Bundle/ProductBundle/Resources/config/services/integrations/doctrine/orm.xml]: You're right 👍

Comment (loic425) [reply src/Sylius/Bundle/ProductBundle/Resources/config/services/integrations/doctrine/orm.xml]: @lchrusciel @GSadee Ok with that ?

Comment (lchrusciel) [reply src/Sylius/Bundle/ProductBundle/Resources/config/services/integrations/doctrine/orm.xml]: we should :)

Comment (loic425) [reply src/Sylius/Bundle/ProductBundle/Resources/config/services/integrations/doctrine/orm.xml]: @lchrusciel @Zales0123 
We forgot something... 
[This line](https://github.com/Sylius/SyliusGridBundle/blob/dfc5f4a0920010550b1af60801179bea7ada5632/src/Bundle/Doctrine/ORM/Driver.php#L48) returns EntityRepository.

However 
```
bin/console debug:container | grep repository.product_association_type
```    
sylius.repository.product_association_type                                                              Sylius\Bundle\ProductBundle\Doctrine\ORM\ProductAssociationTypeRepository                                                      
**sylius.repository.product_association_type_translation**                                                                 Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository

Comment (loic425) [reply src/Sylius/Bundle/ProductBundle/Resources/config/services/integrations/doctrine/orm.xml]: Also, this is the right repository on ResourceController.

**Pull request closed without merging**